### PR TITLE
Fixes package, GAV, protocol, repo name and adding tests.

### PR DESCRIPTION
**Pull request by rhusar**

### CODE REVIEW

File: src/test/resources/org/jgroups/protocols/aws/tpc-NATIVE_S3_PING.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<!--
```
Comment (tristantarrant) [inline src/test/resources/org/jgroups/protocols/aws/tpc-NATIVE_S3_PING.xml]: Rename the file from `tpc` to `tcp`

File: src/test/resources/org/jgroups/protocols/aws/tpc-S3_PING.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
+<!--
```
Comment (tristantarrant) [inline src/test/resources/org/jgroups/protocols/aws/tpc-S3_PING.xml]: Rename the file from `tpc` to `tcp`

**Comment by tristantarrant** [comment]
Two minor changes, but the rest LGTM

File: src/test/resources/org/jgroups/protocols/aws/tpc-NATIVE_S3_PING.xml (hunk 1 above)
Comment (rhusar) [reply src/test/resources/org/jgroups/protocols/aws/tpc-NATIVE_S3_PING.xml]: Now that's a funny typo. Fixed.

File: pom.xml [hunk 3]
```diff
@@ -49,15 +49,15 @@
     </licenses>
 
     <scm>
-        <connection>scm:git:git://github.com/jgroups-extras/native-s3-ping.git</connection>
-        <developerConnection>scm:git:ssh://git@github.com/jgroups-extras/native-s3-ping.git</developerConnection>
+        <connection>scm:git:git://github.com/jgroups-extras/jgroups-aws.git</connection>
```
Comment (tristantarrant) [inline pom.xml:52]: Does this mean that the repo will be renamed after this is merged ?

Comment (rhusar) [reply pom.xml:52]: @tristantarrant That's the intention as per the proposal in the email.

Comment (tristantarrant) [reply pom.xml:52]: Great

**Review by pferraro — APPROVED** [review]

File: pom.xml [hunk 4]
```diff
@@ -1,16 +1,16 @@
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
-    <groupId>org.jgroups.aws.s3</groupId>
-    <artifactId>native-s3-ping</artifactId>
+    <groupId>org.jgroups.aws</groupId>
+    <artifactId>jgroups-aws</artifactId>
     <packaging>jar</packaging>
-    <name>native-s3-ping</name>
+    <name>JGroups - AWS</name>
     <version>2.0.1.Final-SNAPSHOT</version>
```
Comment (pruivo) [inline pom.xml:7]: Since we are changing the repository name, maven coordinates, and protocol name, I would suggest incrementing the major version to 3.

Comment (rhusar) [reply pom.xml:7]: @pruivo The fact is that there has never been a usable release –  https://repository.jboss.org/org/jgroups/aws/s3/native-s3-ping/ thus bumping to 3.x is unnecessary.

Also, the intention is to backport the protocol changes to 1.x as well (no need to change the GAV in 1.x branch of this though)

Comment (tristantarrant) [reply pom.xml:7]: Ideally we would also use Maven relocation

Comment (rhusar) [reply pom.xml:7]: @tristantarrant Yes, we could – but that's done in the "old branch" not this branch (`main`).

**Comment by tristantarrant** [comment]
Merged, thanks

File: pom.xml (hunk 3 above)
Comment (rhusar) [reply pom.xml:52]: Done - repo renamed.

File: pom.xml (hunk 4 above)
Comment (rhusar) [reply pom.xml:7]: Opened https://github.com/jgroups-extras/jgroups-aws/issues/172 issue to track relocation pom.

**Pull request merged**